### ISSUE TITLE
Docs for developers

### ISSUE THREAD

**Issue by teotigraphix** [BowlerHatLLC/vscode-nextgenas#6]
Hey, you said chat so here I am.

Since I have 4 applications built with feathers going to be released soon, my primary focus is AS->Mobile SWF development.

Also as you have noted, MXML is my primary UI so, if I could get a pretty detailed write up or discussion about the moving pieces of this I would love to help.

I am using FB for certain reasons and I have always dreamed about being in control of the IDE I use, reasons the Ranodri compiler (IntelliJ Plugin) and a couple Eclipse extensions were written.

I am all to familiar with Falcon and AST. :) I looked through the code and it's the language serve and your Java that I have no idea how that is plugging in.

Anyway, if you have time, I am a pretty capable programmer with this stuff once I get it but I just don't have the hours to sit and try to get it to build, figure out how things connect and such. (I guess I am speaking about debugging a SWF or calling ADB)

Is there any way to hook up custom HTML Ui in VSCode? Like MXML design view is what I am thinking... way down the road but man, to have it for Feathers and FlexJS might be pretty crazy. I am thinking that you would need a browser, was just curious if there was any API IN VSCode to make custom UI/Panes/Panels whatever they call them.

**Comment by joshtynjala** [comment]
> I looked through the code and it's the language serve and your Java that I have no idea how that is plugging in.

I wrote up a quick and dirty overview of some things. I'll probably be editing this document a lot more, and moving some of it into comments in the actual code. Hopefully, it will give you a good foundation, though. Feel free to keep asking questions in this thread and I can clarify things.

https://github.com/BowlerHatLLC/vscode-nextgenas/wiki/Contribute-to-the-extension

I should also add a vscode channel to my NextGenAS slack, and we can chat in real-time there. I'd be happy to invite you. It's me and some Patreon supporters on the general channel, but I'd love to add some project-specific channels for contributors too. If you have a preferred email to send an invite to (such as the one I already know), let me know.

> Is there any way to hook up custom HTML Ui in VSCode?

There's a vscode.previewHtml command that we might be able to use. It's listed in the [Complex Commands API](https://code.visualstudio.com/docs/extensionAPI/vscode-api-commands).

Basically, it can open a new tab to render HTML instead of a text document. According to the docs, you can put links in there that call custom commands that the extension defines. I'm not sure about sending requests to the HTML file from the extension. 

Besides that, I think VSCode tries not to expose too much UI to extensions, at least for now. You can ask it to show an input with a prompt, a list with multiple options to choose from, or dismissable error/warning/info bars. Maybe some kind of file picker too.

**Comment by teotigraphix** [comment]
Yeah, send it to the one you have.

I am going to spend a bit of time getting familiar with things. I have a couple ideas that would increase my work flow with my apps that could benefit the project so i will hash those out in a while.

I'm curious as to your estimate on how long it would take to get the debugger working for SWF.

If I come up with any more questions in the mean time I will tag them on here.

I figured as much with the View, that is ok. 

I'm just glad the 100's of hours I spent on the cross compiler is actually being used. BTW with Apache Flex and Randori(which went back into FlaconJX) I spent about 1000 hours over 6 months, I charted it all. :)

**Comment by joshtynjala** [comment]
> I am going to spend a bit of time getting familiar with things.

Cool. I'm adding some doc comments to the Java source right now, so that will hopefully put some information more in context to help you out.

> I'm curious as to your estimate on how long it would take to get the debugger working for SWF.

I'm planning to dig into that one next. If I can find a way to spawn the fdb debugger executable and use stdin/stdout to communicate with it, it could be super fast. If I need to instantiate Java classes and do things manually, I'll need to do more investigation before I can estimate anything.

> I'm just glad the 100's of hours I spent on the cross compiler is actually being used. BTW with Apache Flex and Randori(which went back into FlaconJX) I spent about 1000 hours over 6 months, I charted it all. :)

Your work is greatly appreciated, Mike! Between you, Adobe, and everyone else that contributed to the compiler, what I'm building now is sitting on top of a huge foundation that I couldn't have done it without. :)

**Comment by teotigraphix** [comment]
Man where does the time go. :)
